posts/2021-02-15-eclipse-transformer.adoc

```diff
@@ -0,0 +1,506 @@
+
+    https://github.com/OpenLiberty/open-liberty/tree/integration/dev/wlp-jakartaee-transform/rules
+
+Then add three command line arguments to the transformer command line invocation:
+
+[source]
+--
+  java -jar org.eclipse.transformer.cli-0.3.0-SNAPSHOT.jar \
+    <target_location> \
+    <output_location> \
+    -tr ${rules}/jakarta-renames.properties \
+    -td ${rules}/jakarta-direct.properties \
+    -tf ${rules}/jakarta-xml-master.properties
+--
+
+Usually, just three additional command line arguments are needed.  There are additional command line arguments for transforming bundle archives, and for more finely selecting which target files are to be transformed, but those additional arguments are usually not needed.
+
+The target location may be a packaged application archive -- EAR, WAR, RAR, or JAR files -- or may be a directory containing the fully or partially expanded contents of one of these archives.  The target can also be a directory containing java source, or a directory containing several application archives.  All of the artifacts in the target location are processed by the transformer.
+
+=== More about transforming server configurations
+
+Transformation of a server configuration is only necessary when java EE package names are exposed in the server configuration.  One case of explosed java EE packages names are JMS activation specification elements.  For example:
+
+[source]
+--
+  <jmsActivationSpec id="SharedSubscriptionWithMsgSel/TestTopic1">
+    <properties.wasJms
+        destinationRef="jms/FAT_TOPIC"
+        destinationType="javax.jms.Topic"
+        subscriptionDurability="DurableShared"
+        clientId="cid1"
+        subscriptionName="DURSUB"/>
+  </jmsActivationSpec>
+--
+
+This must be transformed to:
+
+[source]
+--
+  <jmsActivationSpec id="SharedSubscriptionWithMsgSel/TestTopic1">
+    <properties.wasJms
+        destinationRef="jms/FAT_TOPIC"
+        destinationType="jakarta.jms.Topic"
+        subscriptionDurability="DurableShared"
+        clientId="cid1"
+        subscriptionName="DURSUB"/>
+  </jmsActivationSpec>
+--
+
+== Updating features in server configurations:
+
+When running a Jakarta EE 9 enable application with Open-Liberty, Jakarta EE 9 versions of Open-Liberty features must be specified in the server configuration.
+
+In many cases, Jakarta EE 9 features use the same short names as the corresponding Java EE features.  The Jakarta EE 9 features simply update the feature version.  Here is a complete table of (non-internal) features which were updated by changing just the feature version:
+
+.Java EE to Jakarta EE 9 Feature Updates: Version Only
+|===
+| Java EE 8 feature name | Jakarta EE 9 feature name
+
+| appClientSupport-1.0
+| appClientSupport-2.0
+
+| appSecurity-3.0
+| appSecurity-4.0
+
+| batch-1.0
+| batch-2.0
+
+| batchManagement-1.0
```

Comment:
batchManagement is not a Jakarta EE 9 feature.  It is a value-add Liberty feature.  batchManagement-2.0 is not shipped today in the beta.  We are still deciding on if we will keep batchManagement-1.0 and just do the right thing for both batch 1.0 and 2.0.  Please do not include it in this list.